proto/eth/v2/beacon_lightclient.proto

```diff
@@ -0,0 +1,71 @@
+// Copyright 2023 Prysmatic Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+syntax = "proto3";
+
+package ethereum.eth.v2;
+
+import "proto/eth/ext/options.proto";
+import "proto/eth/v1/attestation.proto";
+import "proto/eth/v1/beacon_block.proto";
+import "proto/eth/v2/version.proto";
+import "proto/eth/v2/sync_committee.proto";
+
+option csharp_namespace = "Ethereum.Eth.V2";
+option go_package = "github.com/prysmaticlabs/prysm/v4/proto/eth/v2;eth";
+option java_multiple_files = true;
+option java_outer_classname = "SyncCommitteeProto";
+option java_package = "org.ethereum.eth.v2";
+option php_namespace = "Ethereum\\Eth\\v2";
+
+// Beacon LightClient API related messages.
+
+message LightClientBootstrap {
+    v1.BeaconBlockHeader header = 1;
+    SyncCommittee current_sync_committee = 2;
+    repeated bytes current_sync_committee_branch = 3 [(ethereum.eth.ext.ssz_size) = "current_sync_committee_branch.depth,32"];
+}
+
+message LightClientUpdate {
+    v1.BeaconBlockHeader attested_header = 1;
+    SyncCommittee next_sync_committee = 2;
+    repeated bytes next_sync_committee_branch = 3 [(ethereum.eth.ext.ssz_size) = "next_sync_committee_branch.depth,32"];
+    v1.BeaconBlockHeader finalized_header = 4;
+    repeated bytes finality_branch = 5 [(ethereum.eth.ext.ssz_size) = "finality_branch.depth,32"];
+    v1.SyncAggregate sync_aggregate = 6;
+    uint64 signature_slot = 7 [(ethereum.eth.ext.cast_type) = "github.com/prysmaticlabs/prysm/v4/consensus-types/primitives.Slot"];
+}
+
+message LightClientFinalityUpdateResponse {
```

Comment:
This type can be defined under `rpc/eth/lightclient` because it doesn't need ssz

Reply:
This type is used in the http streaming api (see [here](https://github.com/polymerdao/prysm/blob/polymer/lightclient-5/beacon-chain/rpc/eth/events/events.go#L217))  and `streamData()` expects a `proto.Message`.

Reply:
I think we should differentiate between proto messages used in events and HTTP response structs that don't need to be proto messages. After all they are used in two different contexts. I would remove the `Response` suffix from protos and create the response structs next to HTTP handlers.

Reply:
I can remove the `Response` suffix, no prob. 

These types are only used in the events and not by the HTTP handlers. The objects that populate the HTTP handlers' responses are already defined in go structs next to the handlers as you initially suggested. See https://github.com/polymerdao/prysm/blob/polymer/lightclient-5/beacon-chain/rpc/eth/light-client/structs.go

Reply:
I ended up replacing the `Response` suffix with `WithVersion`. The idea is that we wanted to follow the spec and have these methods: [`create_light_client_finality_update`](https://github.com/ethereum/consensus-specs/blob/3d235740e5f1e641d3b160c8688f26e7dc5a1894/specs/altair/light-client/full-node.md#create_light_client_finality_update) and [`create_light_client_optimistic_update`](https://github.com/ethereum/consensus-specs/blob/3d235740e5f1e641d3b160c8688f26e7dc5a1894/specs/altair/light-client/full-node.md#create_light_client_optimistic_update). The result is having the structs we have: `LightClientOptimisticUpdate` and `LightClientFinalityUpdate`. Adding a version to these "goes against" the spec. 

The other approach that we considered was deviating a bit from the spec and have more "efficient" types. For example, we could `LightClientUpdate` to represent both final and optimistic updates. Note that as of now, `LightClientUpdate` is a super set of `LightClientOptimisticUpdate` and `LightClientFinalityUpdate`. The single type could be returned from both finality and optimistic endpoints nullifying the correspondent fields in each case. While we are at it, we could throw the version in there too. 

Again, since we wanted to stay closer to the spec we went for the approach proposed in this PR. However, there's an argument to be made in favour of the second approach.